cmake/ProcessOptions.cmake

```diff
@@ -89,6 +89,13 @@ if ( with-defines )
   endforeach ()
 endif ()
 
+# add colored output from gcc
+if ( with-color )
+  if ( with-color STREQUAL "ON" )
+    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdiagnostics-color=auto" )
```

Comment:
This is a GCC specific option, that is only fully supported for GCC >= 4.9. Older versions might choke:

```
g++-4.8 -fdiagnostics-color=auto -Wall -pedantic -c ...
g++-4.8: error: unrecognized command line option '-fdiagnostics-color=auto'
```

I would recommend to not have this as a user option, but always enable it, if possible (colors are cool and useful during coding), e.g. like this:

```cmake
if ( CMAKE_CXX_COMPILER_ID STREQUAL "GNU" )
  if ( NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9")
    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdiagnostics-color=auto" )
  endif ()
endif ()
```

And also do not forget to add the option for the C compiler (the checks are similar):

```cmake
set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fdiagnostics-color=auto" )
```